jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/BuildImageMojo.java

```diff
@@ -156,25 +153,34 @@ private SourceFilesConfiguration getSourceFilesConfiguration() throws MojoExecut
     }
   }
 
-  private void handleRegistryUnauthorizedException(RegistryUnauthorizedException ex)
-      throws MojoExecutionException {
-    MojoExceptionBuilder mojoExceptionBuilder = new MojoExceptionBuilder(ex);
-
-    if (ex.getHttpResponseException().getStatusCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN) {
-      // No permissions to push to target image.
-      String targetImage = registry + "/" + repository + ":" + tag;
-      mojoExceptionBuilder.suggest("make sure your have permission to push to " + targetImage);
-
-    } else if (credentialHelperName == null) {
-      // Credential helper not defined.
-      mojoExceptionBuilder.suggest("set the configuration 'credentialHelperName'");
-
-    } else {
-      // Credential helper probably was not configured correctly or did not have the necessary
-      // credentials.
-      mojoExceptionBuilder.suggest("make sure your credential helper is set up correctly");
+  // TODO: Add more suggestions for various build failures.
+  /** Appends suggestions to build exceptions. */
+  private <T extends Exception> void handleBuildException(T ex) throws MojoExecutionException {
+    String suggestion = null;
```

Comment:
nit: since it's not reused, it could probably just go in the exception catch block?

Reply:
which might be nice, since you can handle the registry unauthorized exception similarly to how you did before? in it's own block?

Reply:
Moved back.